src/silo/prepare_dataset.cpp

```diff
@@ -15,33 +16,19 @@
 
 [[maybe_unused]] void silo::pruneMetadata(
    std::istream& metadata_in,
-   std::istream& sequences_in,
+   silo::FastaReader& sequences_in,
    std::ostream& metadata_out
 ) {
    SPDLOG_INFO("Pruning metadata");
 
-   std::unordered_set<uint64_t> epi_isl_ids;
+   std::unordered_set<std::string> found_primary_keys;
    uint32_t found_sequences_count = 0;
    uint32_t found_metadata_count = 0;
    {
```

Comment:
Do we need that scope?

Reply:
What do you mean?

Reply:
Ah, I removed one curly brace pair that restricted the scope, but left the other one in, because the variable name key would otherwise clash. I would therefore wait for the refactoring of the metadata reading before deciding on this refactor. I would not want to reuse the variable, which would make it more confusing and neither complicate the names too much, because both keys in the metadata and the sequence file semantically represent the same thing

Reply:
Makes sense. :)